core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -1038,19 +1043,40 @@ public static SqlDataTypeSpec convertTypeToSpec(RelDataType type,
     // interval types, multiset types, etc
     assert typeName != null;
 
-    int precision = typeName.allowsPrec() ? type.getPrecision() : -1;
-    // fix up the precision.
-    if (maxPrecision > 0 && precision > maxPrecision) {
-      precision = maxPrecision;
+    SqlTypeNameSpec typeNameSpec;
+    if (isAtomic(type)) {
```

Comment:
If possible, make `typeNameSpec ` final.

Reply:
Thanks for your advice, I've changed my PR.

Reply:
Try `final SqlTypeNameSpec typeNameSpec` to see if it complains.

Reply:
Thanks for your advice, I've changed my PR.